### PR TITLE
Create ExaGO Package

### PR DESCRIPTION
**Pull request by cameronrutherford**
Creating the ExaGO spack package along with release version 1.0.0.

cc @haampie as you were generous to review @ashermancinelli 's previous PRs

cc @pelesh @cnpetra

### CODE REVIEW

File: var/spack/repos/builtin/packages/exago/package.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    def cmake_args(self):
+        args = []
+        spec = self.spec
+
+        args.append("-DEXAGO_RUN_TESTS=ON")
+
+        if '+mpi' in spec:
+            args.append("-DEXAGO_ENABLE_MPI=ON")
+        else:
+            args.append("-DEXAGO_ENABLE_MPI=OFF")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/exago/package.py]:
```suggestion
        args.append(self.define_from_variant('EXAGO_ENABLE_MPI', 'mpi'))
```

File: var/spack/repos/builtin/packages/exago/package.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        if '+cuda' in spec:
+            args.append("-DEXAGO_ENABLE_GPU=ON")
+
+            cuda_arch_list = spec.variants['cuda_arch'].value
+            cuda_arch = cuda_arch_list[0]
+            if cuda_arch != 'none':
+                args.append(
+                    "-DCMAKE_CUDA_ARCHITECTURES={0}".format(cuda_arch))
```
Comment (haampie) [inline var/spack/repos/builtin/packages/exago/package.py:82]: This requires cmake 3.18 but above you require 3.15

**Review by haampie — CHANGES_REQUESTED** [review]

**Comment by haampie** [comment]
Notice that you likely don't need to set `*_DIR` and that would let you simplify a lot of config options to one-liners:

```diff
-         if '+raja' in spec:
-             args.append("-DEXAGO_ENABLE_RAJA=ON")
-             args.append("-Dumpire_DIR='{0}'".format(spec['umpire'].prefix))
-             args.append("-DRAJA_DIR='{0}'".format(spec['raja'].prefix))
-         else:
-             args.append("-DEXAGO_ENABLE_RAJA=OFF")
+         args.append(self.define_from_variant('EXAGO_ENABLE_RAJA', 'raja'))
```

as long as your cmake uses `find_*` it will consider the spack provided search paths (through CMAKE_PREFIX_PATH)

**Comment by ashermancinelli** [comment]
@haampie `self.define_from_variant` looks very cool and helpful - thanks for pointing this out! @CameronRutherford, could you test out our stack after incorporating @haampie 's suggestions?

**Comment by cameronrutherford** [comment]
@haampie I incorporated your suggestions, and they seem to work! Thanks for the feedback, and hopefully this should be good to go.

**Review by haampie — APPROVED** [review]

**Comment by haampie** [comment]
very pretty :relieved:

**Pull request merged**